### DISCUSSION TITLE
Randomize a single simulation run

### DISCUSSION THREAD

**Discussion by bherd** [eleurent/highway-env#426] [Q&A]
Hi, 

is there a way to randomize an environment such that, starting from the same state, it produces different trajectories? I know that setting the random seed followed by calling the reset function creates a randomized **_initial_** environment. But what I want is to manipulate the random seed during a simulation run such that, starting from an arbitrary state, I can spawn off multiple "sub runs". 

```
// Create environment
env = gym.make("highway-fast-v0")
obs = env.reset()

// Load pretrained model
model = DQN.load("highway_dqn/model", env=env)

// Run simulation for, say, 10 steps
for t in range(10):
  action, _ = model.predict(obs, deterministic=True)
  obs, reward, done, info = env.step(int(action))
  
// Now take the final state of the environment, copy it, say, 5 times
envs = [copy.deepcopy(env) for _ in range(5)]

// For each of those newly created environments, change the seed (such that it behaves differently in the future) 
// and run it for another 10 steps
for env in envs:
  env.seed(int(random.random()*1000))
  for t in range(10):
    action, _ = model.predict(obs, deterministic=True)
    obs, reward, done, info = env.step(int(action))
```

Is something like this possible? My impression is that the using the `seed()` function in that way is useless since only the initialization is randomized and not the dynamics at runtime. Is that correct? Is there any other way to achieve the desired functionality? 

Thanks a lot in advance for any comments or suggestions!

**Comment by eleurent** [comment]
Hi, yes something like this is possible. Given an initial state, the trajectories are affected by the potential destinations of vehicles, and their driving style coefficients.

So you can create different future trajetories from the same state by cloning it and changing these parameters, for each vehicle.
In particular, there is already a method `env.randomize_behaviour()` that you can call and which returns a copy of the environment where some of these parameters are randomly changed within some range, for each vehicle. Feel free to take a look:

https://github.com/eleurent/highway-env/blob/abcef8da79dedf0dfdbe2db5b176d6f203366489/highway_env/envs/common/abstract.py#L379

You can also look at the `set_preferred_lane` and `set_route_at_intersection`, `set_vehicle_field` or `call_vehicle_method` above, all of which can be use to edit the vehicles properties (e.g. target speed, target lane) and affect their future trajectories.
